pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -706,6 +724,18 @@ func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.Clu
 		// Uninstall job exists, check it's status and if successful, remove the finalizer:
 		if controllerutils.IsSuccessful(existingJob) {
 			cdLog.Infof("uninstall job successful, removing finalizer")
+			// duration calculates the time elapsed since the uninstall job started for deprovision jobs that have the DeprovisionStartTimeLabel
+			if startTime, ok := existingJob.Labels[install.DeprovisionStartTimeLabel]; ok {
+				convertedTime, err := strconv.ParseInt(startTime, 10, 64)
+				if err != nil {
+					cdLog.Errorf("Could not parse the unix timestamp")
+				} else {
+					currentTime := time.Now().Unix()
+					duration := currentTime - convertedTime
+					cdLog.Infof("uninstall job duration:%v", duration)
```

Comment:
You should use the job.Status.StartTime and CompletionTime, currentTime may not be right as the controllers might be down for some time, and we are getting this event later on.

cdLog.WithField("duration", jobDuration).Debug("uninstall job completed")